commit e3868984e84b87c606ee7db419d7a189a1278bc3
Author: Morten Linderud <morten@linderud.pw>
Date:   Thu Oct 6 12:00:51 2022 +0200

    strip: fix unique source paths
    
    `${pkgbase}` was added to the wrong invocation. This ensures we are
    producing correct debug packages.
    
    Example from the package:
    
        /usr/src/debug/pacman/pacman-6.0.2/src/pacman/callback.c
        /usr/src/debug/pacman/pacman-6.0.2/src/pacman/callback.h
        /usr/src/debug/pacman/pacman-6.0.2/src/pacman/check.c
        /usr/src/debug/pacman/pacman-6.0.2/src/pacman/check.h
    
    Fixes: 776b7c1e75f2 ("debugflags: Ensure we have unique source paths")
    Signed-off-by: Morten Linderud <morten@linderud.pw>

diff --git a/scripts/libmakepkg/tidy/strip.sh.in b/scripts/libmakepkg/tidy/strip.sh.in
index 91c66f13..01af6e03 100644
--- a/scripts/libmakepkg/tidy/strip.sh.in
+++ b/scripts/libmakepkg/tidy/strip.sh.in
@@ -47,7 +47,7 @@ source_files() {
 
 	LANG=C debugedit --no-recompute-build-id \
 		--base-dir "${srcdir}" \
-		--dest-dir "${dbgsrcdir}/${pkgbase}" \
+		--dest-dir "${dbgsrcdir}" \
 		--list-file /dev/stdout "$1" \
 		| sort -zu | tr '\0' '\n'
 }
@@ -141,7 +141,7 @@ tidy_strip() {
 		if check_option "debug" "y"; then
 
 			dbgdir="$pkgdirbase/$pkgbase-@DEBUGSUFFIX@/usr/lib/debug"
-			dbgsrcdir="${DBGSRCDIR:-/usr/src/debug}"
+			dbgsrcdir="${DBGSRCDIR:-/usr/src/debug}/${pkgbase}"
 			dbgsrc="$pkgdirbase/$pkgbase-@DEBUGSUFFIX@$dbgsrcdir"
 			mkdir -p "$dbgdir" "$dbgsrc"
 		fi
